src/main/java/org/wso2/siddhi/plugins/idea/codeinsight/daemon/impl/SiddhiProjectSdkSetupValidator.java

```diff
@@ -0,0 +1,108 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wso2.siddhi.plugins.idea.codeinsight.daemon.impl;
+
+import com.intellij.codeInsight.daemon.ProjectSdkSetupValidator;
+import com.intellij.openapi.application.WriteAction;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleUtilCore;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.project.ProjectBundle;
+import com.intellij.openapi.projectRoots.Sdk;
+import com.intellij.openapi.roots.ModuleRootManager;
+import com.intellij.openapi.roots.ModuleRootModificationUtil;
+import com.intellij.openapi.roots.ProjectRootManager;
+import com.intellij.openapi.roots.ui.configuration.ProjectSettingsService;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiManager;
+import org.jetbrains.annotations.Nullable;
+import org.wso2.siddhi.plugins.idea.SiddhiFileType;
+import org.wso2.siddhi.plugins.idea.SiddhiLanguage;
+import org.wso2.siddhi.plugins.idea.sdk.SiddhiSdkService;
+import org.wso2.siddhi.plugins.idea.sdk.SiddhiSdkType;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Validates the sdk setup of the project.
+ */
+public class SiddhiProjectSdkSetupValidator implements ProjectSdkSetupValidator {
+
+    public static final SiddhiProjectSdkSetupValidator INSTANCE = new SiddhiProjectSdkSetupValidator();
+
+    @Override
+    public boolean isApplicableFor(@Nonnull Project project, @Nonnull VirtualFile file) {
+        if (file.getFileType() == SiddhiFileType.INSTANCE) {
+            return true;
+        }
+        PsiFile psiFile = PsiManager.getInstance(project).findFile(file);
+        return psiFile != null && psiFile.getLanguage().isKindOf(SiddhiLanguage.INSTANCE);
+    }
+
+    @Nullable
+    @Override
+    public String getErrorMessage(@Nonnull Project project, @Nonnull VirtualFile file) {
+        final Module module = ModuleUtilCore.findModuleForFile(file, project);
+        if (module != null && !module.isDisposed()) {
```

Comment:
Invert the logic to return from method sooner. Fix in other places as well.

Reply:
Fixed in commit 3aa87a6